2012-03-31  Martin Robinson  <mrobinson@igalia.com>

        [GTK] [Stable] --disable geolocation broken after recent merges
        https://bugs.webkit.org/show_bug.cgi?id=82452

        No review, as this is just a build fix.

        Fix the geolocation build.

        * webkit/webkitgeolocationpolicydecision.cpp:
        (webkit_geolocation_policy_decision_new): When gelocation is off, just return null.
        * webkit/webkitgeolocationpolicydecisionprivate.h: Activate webkit_geolocation_policy_decision_new
        for non-Geolocation builds.

Index: /releases/WebKitGTK/webkit-1.8/Source/WebKit/gtk/webkit/webkitgeolocationpolicydecisionprivate.h
===================================================================
--- /releases/WebKitGTK/webkit-1.8/Source/WebKit/gtk/webkit/webkitgeolocationpolicydecisionprivate.h	(revision 112120)
+++ /releases/WebKitGTK/webkit-1.8/Source/WebKit/gtk/webkit/webkitgeolocationpolicydecisionprivate.h	(revision 112800)
@@ -26,6 +26,4 @@
 #include "webkitgeolocationpolicydecision.h"
 
-#if ENABLE(GEOLOCATION)
-
 extern "C" {
 
@@ -34,5 +32,3 @@
 }
 
-#endif // ENABLE(GEOLOCATION)
-
 #endif
Index: /releases/WebKitGTK/webkit-1.8/Source/WebKit/gtk/webkit/webkitgeolocationpolicydecision.cpp
===================================================================
--- /releases/WebKitGTK/webkit-1.8/Source/WebKit/gtk/webkit/webkitgeolocationpolicydecision.cpp	(revision 112120)
+++ /releases/WebKitGTK/webkit-1.8/Source/WebKit/gtk/webkit/webkitgeolocationpolicydecision.cpp	(revision 112800)
@@ -54,8 +54,8 @@
 }
 
-#if ENABLE(GEOLOCATION)
 WebKitGeolocationPolicyDecision* webkit_geolocation_policy_decision_new(WebKitWebFrame* frame, WebCore::Geolocation* geolocation)
 {
-    g_return_val_if_fail(frame, NULL);
+#if ENABLE(GEOLOCATION)
+    g_return_val_if_fail(frame, 0);
     WebKitGeolocationPolicyDecision* decision = WEBKIT_GEOLOCATION_POLICY_DECISION(g_object_new(WEBKIT_TYPE_GEOLOCATION_POLICY_DECISION, NULL));
     WebKitGeolocationPolicyDecisionPrivate* priv = decision->priv;
@@ -64,6 +64,8 @@
     priv->geolocation = geolocation;
     return decision;
+#else
+    return 0;
+#endif
 }
-#endif
 
 /**
 
